### PR TITLE
[dev] newt: enable subpackage python3-newt

### PR DESCRIPTION
**Pull request by mfrw**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
It add a subpackage `python3-newt` for pkg `newt`.

###### Change Log  

- Enable python3-newt

###### Does this affect the toolchain?  

NO

###### Associated issues  

- NA

###### Links to CVEs  
- NA

###### Test Methodology

- local build

### CODE REVIEW

File: SPECS/newt/newt.spec [hunk 1]
```diff
@@ -30,13 +31,26 @@ Requires:	%{name} = %{version}
 %description	devel
 It contains the libraries and header files to create applications
 
+%package -n python3-newt
+%{?python_provide:%python_provide python3-newt}
```
Comment (PawelWMS) [inline SPECS/newt/newt.spec:35]: What's the point of this macro?

File: SPECS/newt/newt.spec [hunk 2]
```diff
@@ -30,13 +31,26 @@ Requires:	%{name} = %{version}
...
+# Remove before F30
```
Comment (PawelWMS) [inline SPECS/newt/newt.spec:36]: Unnecessary comment in our case.

File: SPECS/newt/newt.spec [hunk 3]
```diff
@@ -61,22 +75,37 @@ make %{?_smp_mflags} test
...
-*   Sat Apr 15 2017 Priyesh Padmavilasom <ppadmavilasom@vmware.com> 0.52.20-1
--   Update to 0.52.20
-*   Mon Oct 04 2016 ChangLee <changLee@vmware.com> 0.52.18-3
--   Modified %check
-*   Tue May 24 2016 Priyesh Padmavilasom <ppadmavilasom@vmware.com> 0.52.18-2
--   GA - Bump release of all rpms
-*   Tue Oct 27 2015 Mahmoud Bassiouny <mbassiouny@vmware.com>
--   Initial build.	First version
+* Tue Jul 13 2021 Muhammad Falak R Wani <mwani@microsoft.com> 0.52.21-2
+- Enable python3-newt
```
Comment (PawelWMS) [inline SPECS/newt/newt.spec:86]: Not sure what's the guidelines for extending disto X specs with bits from distro Y, so to be one the safe side let's also mention where we got the new bits from with something like this:

`Extending using Fedora 33 spec (license: MIT).`

File: SPECS/newt/newt.spec [hunk 4]
```diff
@@ -30,13 +31,26 @@ Requires:	%{name} = %{version}
 %description	devel
 It contains the libraries and header files to create applications
 
+%package -n python3-newt
+%{?python_provide:%python_provide python3-newt}
+# Remove before F30
+Provides: %{name}-python3 = %{version}-%{release}
+Provides: %{name}-python3%{?_isa} = %{version}-%{release}
+Provides: snack = %{version}-%{release}
+Summary: Python 3 bindings for newt
```
Comment (PawelWMS) [inline SPECS/newt/newt.spec:40]: Please move the `Summary` tag before all the `Provides` ones.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

File: SPECS/newt/newt.spec (hunk 3 above)
Comment (mfrw) [reply SPECS/newt/newt.spec:85]: Sure :)

File: SPECS/newt/newt.spec (hunk 1 above)
Comment (mfrw) [reply SPECS/newt/newt.spec]: I think it is `Provides: python3-newt`, i've removed it as we already have a package for it.

File: SPECS/newt/newt.spec [hunk 5]
```diff
@@ -61,22 +73,38 @@ make %{?_smp_mflags} test
...
-*   Sat Apr 15 2017 Priyesh Padmavilasom <ppadmavilasom@vmware.com> 0.52.20-1
--   Update to 0.52.20
-*   Mon Oct 04 2016 ChangLee <changLee@vmware.com> 0.52.18-3
--   Modified %check
-*   Tue May 24 2016 Priyesh Padmavilasom <ppadmavilasom@vmware.com> 0.52.18-2
--   GA - Bump release of all rpms
-*   Tue Oct 27 2015 Mahmoud Bassiouny <mbassiouny@vmware.com>
--   Initial build.	First version
+* Tue Jul 13 2021 Muhammad Falak R Wani <mwani@microsoft.com> 0.52.21-2
+- Extend using Fedora 32 spec (license: LGPLv2)
```
Comment (PawelWMS) [inline SPECS/newt/newt.spec:84]: I didn't clarify: the license here indicates how the spec file is distributed, not the package itself. Fedora allows to use its specs under the MIT license - please update.

**Review by PawelWMS — CHANGES_REQUESTED** [review]

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**